python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -246,6 +244,19 @@ class LogisticRegression(Base,
 
         loss = "sigmoid"
 
+        if class_weight is not None:
+            if class_weight == 'balanced':
+                self.class_weight_ = 'balanced'
+            else:
+                classes = list(class_weight.keys())
```

Comment:
Should we save the classes here to check later that the user is indeed passing the classes we expect?

Reply:
I don't think that we need to. If not given, all classes are supposed to have weight one. The `sample_weight *= class_weight[out]` operation should not crash thanks to earlier code that fills ones when necessary :
```
n_explicit = self.class_weight_.shape[0]
if n_explicit != self._num_classes:
    class_weight = cp.ones(self._num_classes)
    class_weight[:n_explicit] = self.class_weight_
    class_weight = CumlArray(class_weight)
    self.class_weight_ = class_weight
```

Reply:
Indeed, but my comment applies more so that the classes that the dictionary of weights has (as keys) have to coincide with the classes of the operation `self.classes_ = cp.unique(y_m)`, no? i.e  if the user passes a class in `fit` that was not in the dict of `class_weights` (when it was passed as a dict) it would be an error, which would be consistent with what Scikit does if I’m not mistaken https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/utils/class_weight.py#L67